sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -127,6 +127,11 @@ namespace Azure { namespace Storage { namespace Blobs {
     constexpr int64_t MaxBlockNumber = 50000;
     constexpr int64_t BlockGrainSize = 1 * 1024 * 1024;
 
+    if (static_cast<uint64_t>(options.TransferOptions.SingleUploadThreshold)
+        > std::numeric_limits<size_t>::max())
+    {
+      throw Azure::Core::RequestFailedException("Single upload threshold is too big");
```

Comment:
What happens if it is set to a negative number? The exception message won't be accurate.

Reply:
I don't think negative value makes any sense. So it'll be undefined behavior. If the customer doesn't provide a valid value, we don't guarantee what would happen.